src/aging_session/memory.clj

```diff
@@ -43,6 +64,67 @@
     (assoc session-map key entry)
     (dissoc session-map key)))
 
+(def ^:private internals-atom (atom {:enc-key nil
+                                     :mac-key nil
+                                     :counter 0}))
+
+(defn get-crypto-keys
+  []
+  (let [{enc-key :enc-key
+         mac-key :mac-key} @internals-atom]
+    (if (and enc-key mac-key)                               ; if keys are present - use them
+      [enc-key mac-key]
+      (do                                                   ; otherwise generate new keys and attempt to set them
+        (swap! internals-atom update-in [:enc-key] #(if-not % (nonce/random-bytes 32) %))
+        (swap! internals-atom update-in [:mac-key] #(if-not % (nonce/random-bytes 128) %))
+        (get-crypto-keys)))))                               ; retrieve the keys again, as something else may have set them in the meantime
+
+(defn decrypt
```

Comment:
Also: can you extract the logic for creating the cookie to another namespace? It seems it's better in terms of separation of concerns

Reply:
will do